lib/timers.js

```diff
@@ -508,17 +508,49 @@ Timeout.prototype.close = function() {
 };
 
 
-var immediateQueue = L.create();
+function ImmediateList() {
+  this.head = null;
+  this.tail = null;
+}
+ImmediateList.prototype.append = function(item) {
+  if (this.tail) {
+    this.tail._idleNext = item;
+    item._idlePrev = this.tail;
+  } else {
+    this.head = item;
+  }
+  this.tail = item;
+};
+ImmediateList.prototype.remove = function(item) {
+  if (item._idleNext) {
+    item._idleNext._idlePrev = item._idlePrev;
+  }
+
+  if (item._idlePrev) {
+    item._idlePrev._idleNext = item._idleNext;
+  }
+
+  if (item === this.head)
+    this.head = item._idleNext;
+  if (item === this.tail)
+    this.tail = item._idlePrev;
+
+  item._idleNext = null;
+  item._idlePrev = null;
+};
+
+
+var immediateQueue = new ImmediateList();
 
 
 function processImmediate() {
-  const queue = immediateQueue;
-  var domain, immediate;
+  var immediate = immediateQueue.head;
+  var tail = immediateQueue.tail;
+  var domain;
 
-  immediateQueue = L.create();
+  immediateQueue.head = immediateQueue.tail = null;
```

Comment:
What did you have in mind? To me it's self-explanatory, it's effectively clearing the linked list.